src/tests/suite/dfuse_test.c

```diff
@@ -474,35 +492,64 @@ do_directory(void **state)
 		assert_return_code(rc, errno);
 	}
 
+	rc = scandirat(dfd, ".", &namelist, NULL, alphasort);
+	assert_int_equal(rc, 100);
+	assert_true(strcmp(namelist[0]->d_name, "file_00") == 0);
+	assert_true(strcmp(namelist[99]->d_name, "file_99") == 0);
+
+	/* free namelist */
+	while (rc--) {
+		free(namelist[rc]);
+	}
+	free(namelist);
+
 	printf("Checking file count\n");
 	dirp = fdopendir(dfd);
 	if (dirp == NULL)
 		assert_return_code(-1, errno);
 
 	pos = telldir(dirp);
 
-	i     = 0;
 	errno = 0;
-	while ((ent = readdir(dirp)) != NULL) {
-		i++;
-	}
+	rc    = get_dir_num_entry(dirp);
 	if (errno != 0)
 		assert_return_code(-1, errno);
-	printf("File count is %d\n", i);
-	assert_int_equal(i, 100);
+	printf("File count is %d\n", rc);
+	assert_int_equal(rc, 100);
 
 	printf("Rewinding and rechecking file count\n");
 	seekdir(dirp, pos);
 
-	i     = 0;
 	errno = 0;
-	while ((ent = readdir(dirp)) != NULL) {
-		i++;
-	}
+	rc    = get_dir_num_entry(dirp);
 	if (errno != 0)
 		assert_return_code(-1, errno);
-	printf("File count is %d\n", i);
-	assert_int_equal(i, 100);
+	printf("File count is %d\n", rc);
+	assert_int_equal(rc, 100);
+
+	env_ldpreload = getenv("LD_PRELOAD");
+	if (env_ldpreload != NULL) {
+		if (strstr(env_ldpreload, "libpil4dfs.so") != NULL) {
+			/* fuse fails in the following test. */
+			for (i = 0; i < 102; i++) {
+				rewinddir(dirp);
+				seekdir(dirp, i);
```

Comment:
This isn't necessarily a problem but the option to seekdir is not a index, it can be any number so this code is not valid for POSIX systems and I'm not surprised it fails on fuse.

This test code in general is run so that it can run against a "standard" filesystem, typically `/tmp` and I'd expect this test to fail if run in that scenario.

Reply:
We set OFFSET_BASE as 2 but do not return "." and ".." in dfuse. This causes the inconsistency with a "standard" filesystem. What are the issues if we return "." and ".." for offset 0 and 1?

Reply:
You're missing the point, `0` and `1` are not meaningful here, seekdir does not take an index, it has to seek to a position that was previously reached by `readdir` and then the position queried by `telldir`.  For `pil4dfs` then 0 and 1 might be valid but they are not in the general case, they won't be for either dfuse or any "standard" filesystem.

I'll push a PR to fix the issue with the test.

Reply:
You need to consider them as "tokens", not positions.  They'll often be inode numbers but the point is they're given to you by the filesystem and the actual values are not meaninful.